### PR TITLE
Add missing square in RBF kernel gradient. Fixes #921

### LINKED ISSUE #921: Missing square in RBF kernel gradient
The following code snippet from the adversarial robustness toolbox computes [the gradient of an RBF kernel](https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/4ca921bb336ec596a8e44f2d5b699df8fbaa9ae9/art/estimators/classification/scikitlearn.py#L1268):

```python
grad = (
    2
    * self.model._gamma
    * (-1)
    * np.exp(-self.model._gamma * np.linalg.norm(x_sample - sv, ord=2))
    * (x_sample - sv)
)
```

According to the [scikit-learn docs](https://scikit-learn.org/stable/modules/svm.html#svm-kernels), the RBF kernel is defined as `\exp(-\gamma \|x-x'\|^2)`. Using the same definition for the RBF kernel, Section 4.2.1 in [this paper by Biggio et al.](https://pralab.diee.unica.it/sites/default/files/biggio14-svm-chapter.pdf), which is also referenced in the code, describes the corresponding gradient as `-2\gamma \exp(-\gamma \| x - x' \|^2) (x - x')`.

Shouldn't the norm in the code be squared?

### PR DESCRIPTION
**Pull request by btlorch**
# Description

Fixes a missing square in the gradient of the RBF kernel

Fixes #921 

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [x] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

I ran the toy example in *classifier_scikitlearn_SVC_LinearSVC.ipynb* and found the result of the projected gradient descent attack to match the findings in [this paper](https://pralab.diee.unica.it/sites/default/files/biggio14-svm-chapter.pdf).

**Test Configuration**:
- Ubuntu 20.04
- Python 3.8
- ART 1.5.2

# Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by beat-buesser** [comment]
Hi @btlorch Thank you very much for your pull request!
